internal/certrotation/var.go

```diff
@@ -0,0 +1,47 @@
+package certrotation
+
+import (
+	"fmt"
+)
+
+const (
+	// CertificateNotBeforeAnnotation contains the certificate expiration date in RFC3339 format.
+	CertificateNotBeforeAnnotation = "tempo.grafana.com/certificate-not-before"
+	// CertificateNotAfterAnnotation contains the certificate expiration date in RFC3339 format.
+	CertificateNotAfterAnnotation = "tempo.grafana.com/certificate-not-after"
+	// CertificateIssuer contains the common name of the certificate that signed another certificate.
+	CertificateIssuer = "tempo.grafana.com/certificate-issuer"
+	// CertificateHostnames contains the hostnames used by a signer.
+	CertificateHostnames = "tempo.grafana.com/certificate-hostnames"
+)
+
+const (
+	// CAFile is the file name of the certificate authority file.
+	CAFile = "service-ca.crt"
+)
+
+// SigningCASecretName returns the microservices signing CA secret name.
+func SigningCASecretName(stackName string) string {
+	return fmt.Sprintf("tempo-%s-signing-ca", stackName)
+}
+
+// CABundleName returns the microservices ca bundle configmap name.
+func CABundleName(stackName string) string {
+	return fmt.Sprintf("tempo-%s-ca-bundle", stackName)
+}
+
+// ComponentCertSecretNames returns a list of all tempo component certificate secret names.
+func ComponentCertSecretNames(stackName string) []string {
+	return []string{
+		fmt.Sprintf("tempo-%s-distributor-http", stackName),
```

Comment:
What's the reason for creating a different cert for each port of each service?
I was looking at the loki operator, and it creates a new service for each port (https://github.com/grafana/loki/blob/61c02d78f2f54c00c6aa47943c4944dc7b0619a7/operator/internal/manifests/distributor.go#L153-L208), but I don't know why.

Reply:
I think the idea is to mantain the HTTP and gRPC TLS configurations separate, as int could be enabled or disables by independent flags. but on the other side, you're right, may be we can use the same certs. I need to think a little bit more on this.